app/Http/Controllers/TaskController.php

```diff
@@ -2,34 +2,48 @@
 
 namespace App\Http\Controllers;
 
+use Auth;
+use App\Models\User;
+use App\Models\Task;
 use Illuminate\Http\Request;
-
+use App\Models\TaskCategory;
+use App\Http\Requests\TaskStoreRequest;
 use Illuminate\Support\Facades\Validator;
 
-use App\Models\Task;
-
 class TaskController extends Controller
 {
     /**
      * Display a listing of the resource.
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-        $taskList = Task::all();
-        return view('Task.dashboard',compact('taskList'));
+        if (isset($request->filter_id)) {
+            $tasks = Task::with('user')->where(['user_id' => Auth::user()->id, 'category_id' => $request->filter_id])->get();
+        } else {
+            $tasks = Task::with('user')->where('user_id', Auth::user()->id)->get();
+        }
+
+        $user = User::where('id', Auth::user()->id)->with('taskCategories')->get();
+        $categories = $user[0]->taskCategories;
+
+        return view('task.dashboard', compact('tasks', 'categories'));
     }
 
-    public function changeStatus(Request $request) {
+    public function toggleCompleted(Request $request)
+    {
         $task = Task::find($request->id);
```

Comment:
Task should be passed in the URL and collected as the argument of `toggleCompleted` method using route model binding.